### ISSUE TITLE
Multiple CodeBuild Dashboard links with same URL

### ISSUE THREAD

**Issue by brokenthumbs** [awslabs/aws-codebuild-jenkins-plugin#27]
**Background:**
I'm using CodeBuild from Jenkins Pipeline, invoking CodeBuild in parallel. Each CodeBuild run executes in parallel, and adds a link to the sidebar to show CodeBuild results inside Jenkins.

```
parallel (
  unitTests: { awsCodeBuild [...], projectName: "unitTests", sourceVersion: "${GIT_COMMIT}" },
  integrationTests: { awsCodeBuild [...], projectName: "integrationTests", sourceVersion: "${GIT_COMMIT}"  }
)
```

**Expected:**
One CodeBuild link to show results from `unitTests` and another CodeBuild link to show results from `integrationTests`

**Actual:**
Multiple CodeBuild links with same URLs. Both of the links lead to `unitTests` results.

![image](https://cloud.githubusercontent.com/assets/1063045/26417769/1b80eefa-4088-11e7-8a2d-15b92de14881.png)

Each of the links above are tied to https://jenkins.com/job/aws-codebuild-job/123/codebuild

---

**Notes:**

Is there any way to make the different links have an additional URI, to separate the links with different AWS CodeBuild results? Something like:

- https://jenkins.com/job/aws-codebuild-job/123/codebuild/unitTests
- https://jenkins.com/job/aws-codebuild-job/123/codebuild/integrationTests

**Comment by leobaran-aws** [comment]
This is a great request, thanks for bringing it to our attention. I'll add it to our queue.

**Comment by uttamk** [comment]
We too faced the same issue when there are multiple parallel steps in a pipeline. We have multiple parallel steps (around 7). Assuming this issue gets fixed and we have different links for different steps and a way to distinguish them visually, I still feel it is a cognitive overhead to 
1. Figure out which step failed
2. Go to the right dashboard link
3. Go to the cloudwatch logs and find out why your build really failed
Ideally, I would love to have the cloudwatch logs in the logs for the step. A quick intermediate solution is to have the cloudwatch logs link in the logs for each step. I have [raised a Pull request for it](https://github.com/awslabs/aws-codebuild-jenkins-plugin/pull/30) for it.

**Comment by leobaran-aws** [comment]
Fixed with cd8065a564520a254f539d4fca00843a339baced. Thanks for bringing this to our attention!
